kernel/src/process.rs

```diff
@@ -771,6 +785,15 @@ impl Process<'a> {
             && buf_end_addr <= self.mem_break()
```

Comment:
Shouldn't buf_end_addr be checked against the application break instead of the kernel break? In other words, what would prevent a grant be allocated in memory which was already allowed?

Reply:
Fixed. `mem_break()` is a hold over from before there was a kernel break and the name is just confusing now. Removed it and replaced it with an explicit use of the right break (`app_break`)